plugins/qtosgrave/osgrenderitem.cpp

```diff
@@ -132,6 +132,100 @@ class WorldCoordOfNodeVisitor : public osg::NodeVisitor
     bool done;
 };
 
+// Encapsulate the Inventor rendering of an Item
+void OSGLODLabel::GlobalLOD::traverse(osg::NodeVisitor& nv)
+{
+    // The input node visitor, represented as a CullStack
+    osg::CullStack *cs;
+    // Use global distance instead of local distance from camera if node visitor settings are applicable
+    // Otherwise, use the default LOD node implementation of this function
+    if (nv.getTraversalMode() == osg::NodeVisitor::TRAVERSE_ACTIVE_CHILDREN && _rangeMode==DISTANCE_FROM_EYE_POINT) {
+        float required_range = nv.getDistanceToViewPoint(getCenter(),true);
+        if ((cs = dynamic_cast<osg::CullStack *>(&nv))) {
+            // Use cull stack's model view matrix to obtain world distance to camera
+            osg::RefMatrix* modelView = cs->getModelViewMatrix();
+            required_range = osg::Vec3d((*modelView)(3,0), (*modelView)(3,1), (*modelView)(3,2)).length();
```

Comment:
Also add some comment here saying:
// we are using modelviewmatrix last column as the camera translation in object space, since OSG matrix is transposed, we are accessing (col, row) instead of (row, col) 

;)